Tests/SwiftParserTest/ParserTests.swift

```diff
@@ -114,15 +113,33 @@ public class ParserTests: XCTestCase {
   /// out into the "swift" directory alongside swift-syntax.
   func testSwiftValidationTestsuite() throws {
     try XCTSkipIf(ProcessInfo.processInfo.environment["SKIP_SELF_PARSE"] == "1")
-    let testDir = URL(fileURLWithPath: #file)
-      .deletingLastPathComponent()
-      .deletingLastPathComponent()
-      .deletingLastPathComponent()
+    let testDir = packageDir
       .deletingLastPathComponent()
       .appendingPathComponent("swift")
       .appendingPathComponent("validation-test")
     runParserTests(
       name: "Swift validation tests", path: testDir, checkDiagnostics: false
     )
   }
+
+  func testSelfParsePerfornamce() throws {
```

Comment:
```suggestion
  func testSelfParsePerformance() throws {
```

Reply:
I wonder if it's worth having a concurrent test as well? Possibly with many more iterations.

Reply:
That's a good idea. Added